cmd/osquery-perf/agent.go

```diff
@@ -280,6 +282,26 @@ func (a *agent) config() {
 		return
 	}
 
+	parsedResp := make(map[string]interface{})
+	if err := json.Unmarshal(res.Body(), &parsedResp); err != nil {
+		log.Println("json parse at config:", err)
+		return
+	}
+
+	var scheduledQueries []string
+	for key, val := range parsedResp {
+		if key == "packs" {
+			packDict := val.(map[string]interface{})
+			for packName, pack := range packDict {
+				queriesDict := pack.(map[string]interface{})["queries"].(map[string]interface{})
```

Comment:
You could use a struct with just the fields you care about - worse case, Unmarshaling won't set any of those fields if they are not there (this would work as long as the `packs` key is necessarily an object which may or may not have a `queries` field that is also an object, and that could be defined as `map[string]interface{}` so you can get the keys without type-casting).

Reply:
Great idea. Honestly, I just went for the simplest my mind could spit to get this going xD